store/postgres/user_model.go

```diff
@@ -6,15 +6,15 @@ import (
 	"github.com/odpf/columbus/user"
 )
 
-type User struct {
+type UserModel struct {
 	ID        string    `db:"id"`
 	Email     string    `db:"email"`
 	Provider  string    `db:"provider"`
```

Comment:
I haven't understood the context completely, but if the information we're going to store is only email & provider. Since both email & provider are required in all APIs' headers, then why do we need to store it?

Reply:
the reason we store email is because all of our features will be using `user_id` for reference.
Since we only receive email from shield, we need to map it to Columbus user's id.

Reply:
email is just the `external` user id (we could consider it as user name). The `id` (uuid) in `user` table is internal user id that we are going to use inside columbus.

Reply:
or something like `account_id`, it should be more generic

Reply:
we were also thinking that way, having a more generic term for external user id. But in that case we can't guarantee the external id would be unique. On the other hand, `email` should be unique and we can make it required as for now.

Reply:
we can create a unique constraint for provider+account_id. But yeah, we can assess that option later once we have more specific use case. Maybe `email` is sufficient for now.

Reply:
yes, we can do that, but currently the `provider` field is not really that required. By default we make it `shield`. 
alright, thanks @rahmatrhd for the input.